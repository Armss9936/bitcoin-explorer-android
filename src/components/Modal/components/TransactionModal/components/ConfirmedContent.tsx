/* eslint-disable import/no-extraneous-dependencies */
import { useShallow } from "zustand/react/shallow";
import { View, StyleSheet } from "react-native";
import { formatDate } from "src/utils/formatBlockInfo";
import BoxContainerWithText from "src/components/BoxContainerWithText/BoxContainerWithText";
import useAppStore from "src/stores/App/useAppStore";
import { TransactionType } from "src/components/Transactions/types";

type ConfirmedContentProps = Required<
  Pick<TransactionType, "fee" | "size" | "statusTransaction">
>;

const styles = StyleSheet.create({
  container: {
    gap: 16,
  },
});

export default function ConfirmedContent(props: ConfirmedContentProps) {
  const { fee, size, statusTransaction } = props;
  const { blockHeight, blockTime } = statusTransaction;
  const localization = useAppStore(useShallow((state) => state.localization));

  const { day, month, year, hour, minutes } = formatDate(blockTime);

  return (
    <View style={styles.container}>
      <View
        style={{
          flexDirection: "row",
          justifyContent: "space-between",
        }}
      >
        <BoxContainerWithText
          firstText={`${localization.t("block")} ${blockHeight}`}
          secondText=""
          width="auto"
        />

        <BoxContainerWithText
          firstText={localization.t("confirmed")}
          secondText=""
          width="auto"
        />
      </View>

      <View>
        <BoxContainerWithText
          firstText={localization.t("date_time")}
          secondText={`${day}/${month}/${year} ${hour}:${minutes}`}
          borderStyles={{
            borderBottomEndRadius: 0,
            borderBottomStartRadius: 0,
            borderBottomColor: "#D9D9D9",
          }}
        />
        <BoxContainerWithText
          firstText={localization.t("size")}
          secondText={`${size} B`}
          borderStyles={{
            borderBottomEndRadius: 0,
            borderBottomStartRadius: 0,
            borderTopEndRadius: 0,
            borderTopStartRadius: 0,
            borderBottomColor: "#D9D9D9",
          }}
        />
        <BoxContainerWithText
          firstText={localization.t("fee")}
          secondText={`${(fee / 100000000).toString()} BTC`}
          borderStyles={{
            borderTopEndRadius: 0,
            borderTopStartRadius: 0,
          }}
        />
      </View>
    </View>
  );
}
