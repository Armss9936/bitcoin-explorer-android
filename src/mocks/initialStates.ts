import { TransactionType } from "src/components/Modal/types";

const timeAgoInitialState = {
  hour: 0,
  minutes: 0,
  day: 0,
  month: 0,
  year: 0,
};

export const initialStateBlocks = [
  {
    blockHeight: 1,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 2,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 3,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 4,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
];

export const initialStateTransaction: TransactionType = {
  transactionId: " ",
  size: 0,
  fee: 0,
  inputTransactions: [
    {
      prevout: {
        scriptpubkey_address: " ",
        value: 0,
      },
    },
  ],
  outputTransactions: [
    {
      scriptpubkey_address: " ",
      value: 0,
    },
  ],
  statusTransaction: undefined,
};
