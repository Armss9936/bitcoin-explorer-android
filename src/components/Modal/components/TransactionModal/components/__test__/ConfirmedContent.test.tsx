/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import { initialStateTransaction } from "src/__mocks__/initialStates";
import ConfirmedContent from "../ConfirmedContent";

jest.mock("src/stores/App/useAppStore", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    localization: { t: (key) => key }, // Mock da função t de tradução
    selectedCurrency: "USD", // Moeda mockada
    bitcoinPrice: {
      mempool: { USD: 50000 },
      blockchainInfo: { USD: { buy: 50000 } },
    }, // Preço fictício
  })),
}));

describe("ConfirmedContent", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ConfirmedContent
          fee={initialStateTransaction.fee}
          size={initialStateTransaction.size}
          statusTransaction={{
            confirmed: true,
            blockHeight: 1,
            blockHash: "abc",
            blockTime: 204,
          }}
          lastBlockHeight={1}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
