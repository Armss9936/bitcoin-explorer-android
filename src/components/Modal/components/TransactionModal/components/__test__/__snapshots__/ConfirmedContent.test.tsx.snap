// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmedContent renders correctly 1`] = `
<View
  style={
    {
      "gap": 16,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "maxWidth": 170,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1d2133",
          "borderBottomColor": undefined,
          "borderBottomEndRadius": undefined,
          "borderBottomStartRadius": undefined,
          "borderBottomWidth": 0,
          "borderRadius": 7,
          "borderTopEndRadius": undefined,
          "borderTopStartRadius": undefined,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "paddingHorizontal": 18,
          "paddingVertical": 9,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 15,
            }
          }
        >
          confirmed
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1d2133",
          "borderBottomColor": "#D9D9D9",
          "borderBottomEndRadius": 0,
          "borderBottomStartRadius": 0,
          "borderBottomWidth": 0.1,
          "borderRadius": 7,
          "borderTopEndRadius": undefined,
          "borderTopStartRadius": undefined,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "paddingHorizontal": 18,
          "paddingVertical": 9,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      >
        date_time
      </Text>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 15,
            }
          }
        >
          31/12/1969 21:3
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1d2133",
          "borderBottomColor": "#D9D9D9",
          "borderBottomEndRadius": 0,
          "borderBottomStartRadius": 0,
          "borderBottomWidth": 0.1,
          "borderRadius": 7,
          "borderTopEndRadius": 0,
          "borderTopStartRadius": 0,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "paddingHorizontal": 18,
          "paddingVertical": 9,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      >
        size
      </Text>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 15,
            }
          }
        >
          0 B
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1d2133",
          "borderBottomColor": undefined,
          "borderBottomEndRadius": undefined,
          "borderBottomStartRadius": undefined,
          "borderBottomWidth": 0,
          "borderRadius": 7,
          "borderTopEndRadius": 0,
          "borderTopStartRadius": 0,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "paddingHorizontal": 18,
          "paddingVertical": 9,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      >
        fee
      </Text>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 15,
            }
          }
        >
          0 BTC
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 14,
            }
          }
        >
          $ 0,00
        </Text>
      </View>
    </View>
  </View>
</View>
`;
